concordium-consensus/src-lib/Concordium/External/GRPC2.hs

```diff
@@ -155,7 +157,17 @@ getTokenInfoV2 cptr blockType blockHashPtr tokenIdPtr tokenIdLen outHash outVec
         Left _ -> return $ queryResultCode QRInvalidArgument
         Right tokenId -> do
             res <- runMVR (Q.getTokenInfo bhi tokenId) mvr
-            returnMaybeMessageWithBlock (copier outVec) outHash res
+            case res of
+                Q.BQRBlock _ (Left QTIEUnknownToken) -> do
+                    copyHashTo outHash res
+                    return $ queryResultCode QRNotFound
+                Q.BQRBlock _ (Left e) -> do
```

Comment:
Maybe this shouldn't be a catch all case and instead explicitly use `QTIEInternal` in case we expand the error cases in the future.